presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorIntegrationSmokeTest.java

```diff
@@ -704,4 +704,34 @@ public void testTableStatsSystemTable()
         // cleanup
         assertUpdate("DROP TABLE test_table_stats");
     }
+
+    @Test
+    public void testAlterTable()
+            throws InterruptedException
+    {
+        assertUpdate("CREATE TABLE test_table (c1 bigint, c2 bigint)");
+        assertUpdate("INSERT INTO test_table VALUES (1, 1), (1, 2), (1, 3), (1, 4)", 4);
+        assertUpdate("INSERT INTO test_table VALUES (11, 1), (11, 2)", 2);
+
+        assertUpdate("ALTER TABLE test_table ADD COLUMN c3 bigint");
+        assertQueryFails("ALTER TABLE test_table DROP COLUMN c3", "Cannot drop the column which has the largest column ID in the table");
+        assertUpdate("INSERT INTO test_table VALUES (2, 1, 1), (2, 2, 2), (2, 3, 3), (2, 4, 4)", 4);
+        assertUpdate("INSERT INTO test_table VALUES (22, 1, 1), (22, 2, 2), (22, 4, 4)", 3);
+
+        // Do a partial delete on shard that has an extra column
+        assertUpdate("DELETE FROM test_table WHERE c1=1 and c2=1", 1);
+        // Then drop a full shard
+        assertUpdate("DELETE FROM test_table WHERE c1=11", 2);
+
+        // Drop a center column
```

Comment:
Nit: the word "center" sounds a bit funny here. The usage is technically correct, but "middle" seems better, though I can't articulate why.
```java
// Drop a column from the middle of the table